core/include/traccc/fitting/status_codes.hpp

```diff
@@ -14,6 +14,7 @@ enum class kalman_fitter_status : uint32_t {
     ERROR_QOP_ZERO,
     ERROR_THETA_ZERO,
     ERROR_INVERSION,
+    ERROR_CHI2_NEGATIVE,
```

Comment:
Shall we make it two for better debugging?:
`ERROR_FILTERED_CHI2_NEGATIVE`
`ERROR_SMOOTHED_CHI2_NEGATIVE`

The return values above will need to change accordingly

Reply:
Hmm, sure I can do that. Should we then also change all the other error codes to have corresponding `FILTER` and `SMOOTH` versions?

Reply:
I've added smoother and filter versions of these errors now.

Reply:
> Should we then also change all the other error codes to have corresponding FILTER and SMOOTH versions?

Yes I think that would be better but not for this PR to sort out